File: Library_LJ/src/com/lj/library/util/PreferenceUtil.java
package com.lj.library.util;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.SharedPreferences;

import com.lj.library.bean.UserInfo;

public class PreferenceUtil {

	private final SharedPreferences sp;
	private final SharedPreferences.Editor editor;

	@SuppressLint("CommitPrefEdits")
	public PreferenceUtil(Context context, String file) {
		sp = context.getSharedPreferences(file, Context.MODE_PRIVATE);
		editor = sp.edit();
	}

	public boolean isFirstRun() {
		return sp.getBoolean("isFirstRun", true);
	}

	public void setIsFirstRun(boolean isFirstOpen) {
		editor.putBoolean("isFirstRun", isFirstOpen);
		editor.commit();
	}

	public String getFlashImageUrl() {
		return sp.getString("flashUrl", "");
	}

	public void setFlashImageUrl(String url) {
		editor.putString("flashUrl", url);
		editor.commit();
	}

	public void setUserInfo(UserInfo userInfo) {
		editor.putString("userId", userInfo.userId);
		editor.commit();
	}

	public UserInfo getUserInfo() {
		String userId = sp.getString("userId", "");
		UserInfo userInfo = new UserInfo();
		userInfo.userId = userId;
		return userInfo;
	}

	public void setUsername(String username) {
		editor.putString("username", username);
		editor.commit();
	}

	public String getUsername() {
		return sp.getString("username", "");
	}

	public void setPassword(String password) {
		editor.putString("password", password);
		editor.commit();
	}

	public String getPassword() {
		return sp.getString("password", "");
	}
}

File: app/src/main/java/com/lj/library/util/PreferenceUtil.kt
package com.lj.library.util

import android.annotation.SuppressLint
import android.content.Context
import android.content.SharedPreferences
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey
import com.lj.library.bean.UserInfo

class PreferenceUtil @SuppressLint("CommitPrefEdits") constructor(context: Context, filename: String) {

    private val sp: SharedPreferences
    private val editor: SharedPreferences.Editor

    init {
//        sp = context.getSharedPreferences(filename, Context.MODE_PRIVATE)

        // 使用加密的SharedPreferences
        sp = getEncryptedSharedPreferences(context, filename)
        editor = sp.edit()
    }

    private fun getEncryptedSharedPreferences(context: Context, filename: String): SharedPreferences {

        val masterKey: MasterKey = MasterKey.Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()

        val sharedPreferences = EncryptedSharedPreferences.create(context,
                filename,
                masterKey,
                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
        )

        return sharedPreferences
    }

    var isFirstRun: Boolean
        get() = sp.getBoolean("isFirstRun", true)
        set(isFirstOpen) {
            editor.putBoolean("isFirstRun", isFirstOpen)
            editor.commit()
        }

    var flashImageUrl: String?
        get() = sp.getString("flashUrl", "")
        set(url) {
            editor.putString("flashUrl", url)
            editor.commit()
        }

    var userInfo: UserInfo
        get() {
            val userId = sp.getString("userId", "")
            val userInfo = UserInfo()
            userInfo.userId = userId
            return userInfo
        }
        set(userInfo) {
            editor.putString("userId", userInfo.userId)
            editor.commit()
        }

    var username: String?
        get() = sp.getString("username", "")
        set(username) {
            editor.putString("username", username)
            editor.commit()
        }

    var password: String?
        get() = sp.getString("password", "")
        set(password) {
            editor.putString("password", password)
            editor.commit()
        }


}